demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+	}
+
+	/**
+	 * Tests selection for the dataTable Selection and Server Events testers
+	 *
+	 * @param  componentUseCase
+	 */
+	protected void testDataTableSelection(Browser browser, String componentUseCase) {
+
+		// Test that the Selection mode defaults to 'checkbox'.
+		SeleniumAssert.assertElementVisible(browser,
+			labelSelectOptionValueXpath("Selection Mode", "checkbox'][@selected='selected"));
+
+		// 'checkbox'
+		testFirstCustomerRow(browser, componentUseCase);
+
+		// Test the selected customers.
+		String[] twoCustomers = { defaultCUSTOMERS[0], defaultCUSTOMERS[2] };
+		testCustomersSelected(browser, twoCustomers, componentUseCase);
+
+		// Test all customers on current page selected.
+		testCustomersSelected(browser, defaultCUSTOMERS, componentUseCase);
+
+		// Select 'radio' Selection Mode.
+		browser.click(labelSelectOptionValueXpath("Selection Mode", "radio"));
+		browser.waitForElementVisible("//input[contains(@type,'radio')]");
+		SeleniumAssert.assertElementVisible(browser,
+			getTableRowCustomersXpath("John", "Adams") + "/..//input[contains(@type,'radio')]");
+
+		// 'radio'
+		testFirstCustomerRow(browser, componentUseCase);
+	}
+
+	/**
+	 * Tests that the Server Events fire at the moment of selection. Can be used for 'checkbox' or 'radio'.
+	 *
+	 * @param  rowIndex
+	 * @param  customers
+	 */
+	protected void testDataTableServerEvents(Browser browser, String rowIndex, String[] customers) {
+
+		for (String customer : customers) {
+			String[] tokens = customer.split(" ");
+
+			// Select the customer.
+			browser.click(getTableRowCustomersXpath(tokens[0], tokens[1]));
+
+			// Test that RowSelectEvent and RowDeselectEvent fire.
+			browser.waitForElementVisible(alloyMessageXpath(
+					serverEventInfoText("RowSelectEvent", rowIndex, arrayToString(customers))));
+
+			if (!browser.findElements(By.ByXPath.xpath("//input[contains(@type,'radio')]")).isEmpty()) {
+
+				controlOrCommandClick(browser, getTableRowCustomersXpath(tokens[0], tokens[1]));
+			}
+			else {
+				browser.click(getTableRowCustomersXpath(tokens[0], tokens[1]));
+			}
+
+			browser.waitForElementVisible(alloyMessageXpath(
+					serverEventInfoText("RowDeselectEvent", rowIndex, arrayToString(customers))));
+			browser.click(getTableRowCustomersXpath(tokens[0], tokens[1]));
+			browser.waitForElementVisible(alloyMessageXpath(
+					serverEventInfoText("RowSelectEvent", rowIndex, arrayToString(customers))));
+		}
+
+		// Reset UI state
+		navigateToUseCase(browser, "dataTable", "server-events");
+	}
+
+	/**
+	 * Tests the first default rows of selection. Can be used for 'checkbox' or 'radio'.
+	 *
+	 * @param  componentUseCase
+	 */
+	protected void testFirstCustomerRow(Browser browser, String componentUseCase) {
+
+		// Test the first row selection
+		browser.click(getTableRowCustomersXpath("John", "Adams"));
+		browser.waitForElementVisible(getTableRowCustomersXpath("John", "Adams") +
```

Comment:
Please create a `waitForRow()` method to encapsulate this wait and make it more readable:

```
private void waitForRowSelectedDeselected(browser,  boolean selected, firstName, lastName) {

		String tr = "//tr[not(contains(@class,'info'))]";

		if (selected) {
			tr = "//tr[contains(@class,'info')]";
		}
		
		browser.waitForElementVisible(tr + "[contains(.,'" + firstName + "')][contains(.,'" + lastName + "')]");
	}
```